### ISSUE TITLE
Issue: High Retransmits and dropped packets on Windows 10 64bit

### ISSUE THREAD

**Issue by tommygr** [Haivision/srt#324]
Configuration:  SRT Gateway  ->  Windows 10 box using stransmite.exe to decode to udp -> playback UDP stream on VLC.   

On Windows 10 64bit we are using stransmit.exe to convert to a UDP stream so that we can play back the stream on VLC 64bit.   To see the issue you don't even have to start playback.  You can start stransmit then got to the Media Gateway and watch the statistics.  

This only seems to be happening on Windows 10.  We tried on multiple versions of Windows 10 and the same results for all. This doesn't happen on Windows 7 64bit.  

![image](https://user-images.githubusercontent.com/13950831/38100576-9fef27ae-3332-11e8-88bc-d14ee0de3736.png)

-Tommy Greene
Vulcan Inc.

**Comment by ethouris** [comment]
Would you be able to collect debug logs from stransmit on Windows 10? Just remember to redirect to a file because especially on Windows the console printing will kill the performance completely.

**Comment by tommygr** [comment]
I ran log files for both Win10 and Win7.  For some reason, the win7 log was quite large so I only did 10 sec test.


[win10_logfile.log](https://github.com/Haivision/srt/files/1861305/win10_logfile.log)

[win7_logfile.log](https://github.com/Haivision/srt/files/1861332/win7_logfile.log)

**Comment by tommygr** [comment]
[logfile_win10_debug.log](https://github.com/Haivision/srt/files/1896575/logfile_win10_debug.log)
Here is another log file.  I think this is more what you were looking for.

**Comment by ethouris** [comment]
I have written a small script to analyze the sequence numbers extracted from this log. It simply traces the sequence number and recognizes packet loss, recovery and excessive packets, in case when the packet was "orderly sent", it prints nothing. This is the result:
```
EXCESSIVE packet: 1379017858 (current 1379017860) STILL LOST: 
15:41:57.474581/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379017858 offset=0 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379017858 (current 1379017860) STILL LOST: 
15:41:57.476980/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379017858 offset=0 EXCESSIVE (UNACKED/RXS-UNKNOWN)
PACKET LOSS: 1379019200 (current 1379019196): CURRENTLY: 1379019196 1379019197 1379019198 1379019199
15:42:04.873776/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379019200 offset=4 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379019196 STILL LOST: 1379019196 1379019197 1379019198 1379019199
15:42:04.875833/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379019196 offset=-1 EXCESSIVE (BELATED/RXS-UNKNOWN)
RECOVERED 1379019197 STILL LOST: 1379019197 1379019198 1379019199
15:42:04.876717/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379019197 offset=0 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379019198 STILL LOST: 1379019198 1379019199
15:42:04.882313/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379019198 offset=1 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379019199 STILL LOST: 1379019199
15:42:04.887639/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379019199 offset=2 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379019200 (current 1379019201) STILL LOST: 
15:42:04.892039/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379019200 offset=-2 EXCESSIVE (BELATED/RXS-UNKNOWN)
PACKET LOSS: 1379020751 (current 1379020748): CURRENTLY: 1379020748 1379020749 1379020750
15:42:13.373293/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379020751 offset=2 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379020749 STILL LOST: 1379020748 1379020749 1379020750
15:42:13.379685/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379020749 offset=0 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379020749 (current 1379020752) STILL LOST: 1379020748 1379020750
15:42:13.380946/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379020749 offset=0 EXCESSIVE (UNACKED/RXS-UNKNOWN)
RECOVERED 1379020750 STILL LOST: 1379020748 1379020750
15:42:13.386405/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379020750 offset=1 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379020751 (current 1379020752) STILL LOST: 1379020748
15:42:13.389712/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379020751 offset=2 EXCESSIVE (UNACKED/RXS-UNKNOWN)
PACKET LOSS: 1379021467 (current 1379021464): CURRENTLY: 1379020748 1379021464 1379021465 1379021466
15:42:17.264225/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021467 offset=3 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379021464 STILL LOST: 1379020748 1379021464 1379021465 1379021466
15:42:17.267087/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021464 offset=-1 EXCESSIVE (BELATED/RXS-UNKNOWN)
RECOVERED 1379021465 STILL LOST: 1379020748 1379021465 1379021466
15:42:17.269134/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021465 offset=0 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379021466 STILL LOST: 1379020748 1379021466
15:42:17.272525/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021466 offset=1 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379021467 (current 1379021468) STILL LOST: 1379020748
15:42:17.279061/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021467 offset=2 EXCESSIVE (UNACKED/RXS-UNKNOWN)
PACKET LOSS: 1379021718 (current 1379021715): CURRENTLY: 1379020748 1379021715 1379021716 1379021717
15:42:18.577488/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021718 offset=3 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379021715 STILL LOST: 1379020748 1379021715 1379021716 1379021717
15:42:18.580646/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021715 offset=-1 EXCESSIVE (BELATED/RXS-UNKNOWN)
RECOVERED 1379021716 STILL LOST: 1379020748 1379021716 1379021717
15:42:18.580677/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021716 offset=0 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379021717 STILL LOST: 1379020748 1379021717
15:42:18.588354/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021717 offset=1 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379021718 (current 1379021718) STILL LOST: 1379020748
15:42:18.595338/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379021718 offset=2 EXCESSIVE (UNACKED/RXS-UNKNOWN)
PACKET LOSS: 1379022229 (current 1379022226): CURRENTLY: 1379020748 1379022226 1379022227 1379022228
15:42:21.465990/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379022229 offset=4 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379022225 (current 1379022229) STILL LOST: 1379020748 1379022226 1379022227 1379022228
15:42:21.468004/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379022225 offset=-2 EXCESSIVE (BELATED/RXS-UNKNOWN)
RECOVERED 1379022227 STILL LOST: 1379020748 1379022226 1379022227 1379022228
15:42:21.469745/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379022227 offset=0 ACCEPTED (EXPECTED/RXS-UNKNOWN)
RECOVERED 1379022228 STILL LOST: 1379020748 1379022226 1379022228
15:42:21.473783/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379022228 offset=1 ACCEPTED (EXPECTED/RXS-UNKNOWN)
EXCESSIVE packet: 1379022229 (current 1379022229) STILL LOST: 1379020748 1379022226
15:42:21.478406/SRT:RcvQ:worker D: SRT.c: %895977431:RECEIVED: seq=1379022229 offset=2 EXCESSIVE (UNACKED/RXS-UNKNOWN)
```

For me it doesn't look like very excessive packet loss record.

**Comment by maxlovic** [comment]
Hi @tommygr,
Please check if PR #510 fixes the issue.

**Comment by maxlovic** [comment]
Hi @tommygr, 
We are preparing to release [v.1.3.2](https://github.com/Haivision/srt/commits/v1.3.2-rc.0) that is expected to resolve the issue you have. Please respond.

**Comment by tommygr** [comment]
Thank You.  
I'm on a different project right now and I don't have a way to confirm at this point in time.
I will be able to test in about a week from now.  ( Feb. 1 )

**Comment by maxlovic** [comment]
Closing due to no activity. Potentially resolved by PR #510 in v1.3.2
